hail/src/main/scala/is/hail/expr/ir/agg/Extract.scala

```diff
@@ -68,6 +69,8 @@ object AggStateSig {
     case ArrayAggStateSig(nested) => new ArrayElementState(cb, StateTuple(nested.map(sig => AggStateSig.getState(sig, cb)).toArray))
     case GroupedStateSig(kt, nested) => new DictState(cb, kt, StateTuple(nested.map(sig => AggStateSig.getState(sig, cb)).toArray))
     case NDArraySumStateSig(nda) => new TypedRegionBackedAggState(nda, cb)
+    case NDArrayMultiplyAddStateSig(tupleNDA) =>
+      new TypedRegionBackedAggState(VirtualTypeWithReq.apply(tupleNDA.canonicalPType.asInstanceOf[PCanonicalTuple].types(0).setRequired(false)), cb)
```

Comment:
I was confused by the types of the MultiplyAdd aggregator for a while. I think it would be more straightforward for `NDArrayMultiplyAddStateSig` to take just the single `PNDArray` type, the seqOp to take the left and right ndarrays as two separate arguments, and `NDArrayMultiplyAddAggregator` to also take the left and right NDArray types separately.